beartype_test/a90_func/package/test_package_import.py

```diff
@@ -48,9 +49,12 @@ def test_package_import_isolation() -> None:
         r'beartype\.cave',
         r'beartype\.vale',
         r'numpy',
-        r'typing_extensions',
     )
 
+    # Add typing_extensions to the prohibition list unless we're inside 3.7
+    if not IS_PYTHON_3_7:
+        _HEAVY_MODULE_NAME_RAW_REGEXES += (r'typing_extensions',)
```

Comment:
This is a compromise to address https://github.com/beartype/beartype/pull/103/files#r814527877.